openssl/src/asn1.rs

```diff
@@ -95,6 +114,97 @@ foreign_type_and_impl_send_sync! {
     pub struct Asn1TimeRef;
 }
 
+impl Asn1TimeRef {
+    /// Find difference between two times
+    ///
+    /// This corresponds to [`ASN1_TIME_diff`].
+    ///
+    /// [`ASN1_TIME_diff`]: https://www.openssl.org/docs/man1.1.0/crypto/ASN1_TIME_diff.html
+    #[cfg(ossl102)]
+    pub fn diff(&self, compare: Option<&Self>) -> Result<TimeDiff, ErrorStack> {
```

Comment:
Let's have this just take `compare: &Self` - we can add separate methods for the other case if necessary.